src/ControllerV2/BeneficiaryResetPasswordController.php

```diff
@@ -35,4 +39,51 @@ public function resetEmail(Request $request, Beneficiaire $beneficiary, Resettin
             'beneficiary' => $beneficiary,
         ]);
     }
+
+    #[IsGranted('UPDATE', 'beneficiary')]
+    #[Route(path: '/beneficiary/{id}/reset-password/sms', name: 'reset_password_beneficiary_sms', methods: ['GET', 'POST'])]
+    public function resetSMS(
+        Request $request,
+        Beneficiaire $beneficiary,
+        ResettingService $service,
+        TranslatorInterface $translator,
+    ): Response {
+        $userToReset = $beneficiary->getUser();
+        if (!$phone = $userToReset->getTelephone()) {
+            $this->addFlash('error', 'beneficiary_has_no_phone_number');
+
+            return $this->redirectToRoute('reset_password_beneficiary', ['id' => $beneficiary->getId()]);
+        } elseif ($service->isRequestingByEmail($userToReset)) {
+            $this->addFlash('danger', 'public_reset_password_already_requested');
+            $this->addFlash('danger', 'reset_password_requested_by_email');
+
+            return $this->redirectToRoute('reset_password_beneficiary', ['id' => $beneficiary->getId()]);
```

Comment:
I think we should allow the user to request both method at the same time

Reply:
This feature is based on the symfonycast/reset-password-bundle, for this reason a user can make only one request at a time, otherwise a TooManyPasswordRequestException is thrown.